Client/Frontend/Browser/BrowserLocationView.swift

```diff
@@ -132,13 +138,23 @@ class BrowserLocationView : UIView, UIGestureRecognizerDelegate {
     var url: NSURL? {
         didSet {
             lockImageView.hidden = (url?.scheme != "https")
-            if let url = url?.absoluteString {
-                if url.hasPrefix("http://") ?? false {
-                    locationLabel.text = url.substringFromIndex(advance(url.startIndex, 7))
-                } else if url.hasPrefix("https://") ?? false {
-                    locationLabel.text = url.substringFromIndex(advance(url.startIndex, 8))
+
+            if let urlString = url?.absoluteString {
+
+                var truncatedURL: String
+                if let scheme = url?.scheme {
+                    truncatedURL = urlString.substringFromIndex(advance(urlString.startIndex, count(scheme) + count("://")))
```

Comment:
I think we may want to stick with explicitly checking `http://` here rather than removing all schemes. If a page uses HTTPS, we want it to be visible in the URL bar so it's extra clear to the user (at least, this is what we do on Firefox Android and desktop).

Also, I think it's possible to have a URL with a scheme that doesn't have `://` in the string (e.g., `about:home`).